packages/core/apps/core-apps-server-internal/src/core_app.ts

```diff
@@ -201,7 +295,12 @@ export class CoreAppsService {
         },
         async (context, req, res) => {
           try {
-            this.configService.setDynamicConfigOverrides(req.body);
+            const newGlobalOverrides = this.configService.setDynamicConfigOverrides(req.body);
+            const soClient = await firstValueFrom(savedObjectClient$);
+            await soClient.create(DYNAMIC_CONFIG_OVERRIDES_SO_TYPE, newGlobalOverrides, {
```

Comment:
nit: I guess if `config.allowDynamicConfigOverrides === false` we could skip this write.

Reply:
This API is not registered when the config is false, that's why I didn't add any check here.